uswid/identity.py

```diff
@@ -73,6 +78,24 @@ def entities(self) -> List[uSwidEntity]:
         """returns all the added entities"""
         return list(self._entities.values())
 
+    # get a string representation of tag_id
+    # it's either just the plain tag_id or a hex representation of an UUID
+    def get_tagid_string(self) -> str:
+        if type(self.tag_id) is str:
+            return self.tag_id
+        elif type(self.tag_id) is bytes:
+            return uuid.UUID(bytes=self.tag_id).hex
```

Comment:
i don't get it; why convert it to a UUID at all?

Reply:
> tag-ids are now represented as a 16 byte UUID

Is that in the spec tho? I thought it was an optional text string.

Reply:
> tag-id (index 0): A 16-byte binary string, or a textual identifier, uniquely referencing a software component.

Either it's a 16 byte binary string OR a textual identifier.
The Idea is to look which type the tag-id has in the CBOR format. If it's a byte array, it has to be a 16-byte binary array. Knowing that we try to convert it to a UUID, to check if it's a legit UUID which we can convert to a string. Although I guess any 16-byte binary array is a legit UUID, so we can just as well convert it directly to a string. I will make the change.
If the tag-id is already a string in CBOR format, we don't have to do any kind of conversion to string, since it already is one. I am actually not sure if the 16 bytes are also a constraint if the tag-id is a string (textual identifier), but I guess not.

Reply:
> Either it's a 16 byte binary

Is that in the coSWID spec?

Reply:
That is from the CoSWID spec:
> tag-id (index 0): A 16-byte binary string, or a textual identifier, uniquely referencing a software component.

Reply:
Doh, I was looking at the SWID spec, not the coSWID spec. I think we probably ought to be much more strict about the 16 byte GUID thing, and only store a string internally. I'll fix up your stuff into something a bit more pythonic.

Reply:
_narrator_: he did this.